litho-it/src/test/java/com/facebook/litho/specmodels/model/PropValidationTest.java

```diff
@@ -210,6 +210,20 @@ public void testVarArgPropMustHaveParameterizedListType() {
   }
 
   @Test
+  public void testIncorrectTypeForResTypeWithVarArg() {
+    when(mPropModel1.getResType()).thenReturn(ResType.BOOL);
+    when(mPropModel1.hasVarArgs()).thenReturn(true);
+    when(mPropModel1.getType()).thenReturn(ParameterizedTypeName.get(ClassNames.LIST, TypeName.INT.box()));
+
+    List<SpecModelValidationError> validationErrors = PropValidation.validate(mSpecModel, PropValidation.RESERVED_PROP_NAMES);
+    assertThat(validationErrors).hasSize(1);
+    assertThat(validationErrors.get(0).element).isEqualTo(mRepresentedObject1);
+    assertThat(validationErrors.get(0).message).isEqualTo(
+        "A prop declared with resType BOOL must be one of the following types: " +
```

Comment:
Could you add a note to the error message that this is about a var-arg prop?

Reply:
It's done